crates/ruff_linter/src/noqa.rs

```diff
@@ -424,21 +425,23 @@ impl<'a> FileNoqaDirectives<'a> {
 /// An individual file-level exemption (e.g., `# ruff: noqa` or `# ruff: noqa: F401, F841`). Like
 /// [`FileNoqaDirectives`], but only for a single line, as opposed to an aggregated set of exemptions
 /// across a source file.
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug)]
 pub(crate) enum ParsedFileExemption<'a> {
     /// The file-level exemption ignores all rules (e.g., `# ruff: noqa`).
     All,
     /// The file-level exemption ignores specific rules (e.g., `# ruff: noqa: F401, F841`).
-    Codes(Vec<&'a str>),
+    Codes(Codes<'a>),
 }
 
 impl<'a> ParsedFileExemption<'a> {
     /// Return a [`ParsedFileExemption`] for a given comment line.
-    fn try_extract(line: &'a str) -> Result<Option<Self>, ParseError> {
+    fn try_extract(line: &'a str, offset: TextSize) -> Result<Option<Self>, ParseError> {
+        let init_line_len = line.len();
         let line = Self::lex_whitespace(line);
         let Some(line) = Self::lex_char(line, '#') else {
             return Ok(None);
         };
+        let comment_start = init_line_len - line.len() - '#'.len_utf8();
```

Comment:
Nit: I suggest using `TextLen` methods here to avoid the conversation further down
```suggestion
        let init_line_len = line.text_len();
        let line = Self::lex_whitespace(line);
        let Some(line) = Self::lex_char(line, '#') else {
            return Ok(None);
        };
        let comment_start = init_line_len - line.text_len() - '#'.text_len();
```